### PR TITLE
feat: Added dlpack device

### PR DESCRIPTION
**Pull request by venkatramnank**
## Proposed changes

Isuue #1159 , Added `__dlpack_device__` to MLX array. Followed the values present in dlpack docs: https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__dlpack_device__.html 
Returns `tuple` with `(device type, device id)`.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: python/src/array.cpp [hunk 1]
```diff
@@ -353,6 +355,19 @@ void init_array(nb::module_& m) {
...
+          "__dlpack_device__",
+          [](const array& a) {
+            if (metal::is_available()) {
+              // Metal device is available
+              constexpr int kDLMetal = 8;
+              return kDLMetal;
+            } else {
+              // CPU device
+              constexpr int kDLCPU = 1;
+              return kDLCPU;
```
Comment (awni) [inline python/src/array.cpp:368]:
```suggestion
              return 8;
            } else {
              // CPU device
              return 1;
```

File: docs/src/usage/numpy.rst [hunk 2]
```diff
@@ -58,6 +58,7 @@ The squaring of ``x`` occurs externally to MLX, meaning that no gradient is inco
 It's important to note that a similar issue arises during array conversion and copying.
 For instance, a function defined as ``mx.array(np.array(x)**2).sum()`` would also result in an incorrect gradient,
 even though no in-place operations on MLX memory are executed.
+``mx.array`` supports dlpack as well.
```
Comment (awni) [inline docs/src/usage/numpy.rst:61]: This is not the right spot for this. Too buried. I will make a suggestion further up.

Comment (awni) [reply docs/src/usage/numpy.rst:61]: Can you do something like at Line 6?

```
MLX array supports conversion between other frameworks through either:

1. The `Python Buffer Protocol <https://docs.python.org/3/c-api/buffer.html>`_. 
2. `DLPack <https://dmlc.github.io/dlpack/latest/>`_.

Let's convert an array to NumPy and back.
```

Comment (venkatramnank) [reply docs/src/usage/numpy.rst]: I have made the requested changes. Thank you!!

File: docs/src/usage/numpy.rst [hunk 3]
```diff
@@ -4,6 +4,12 @@ Conversion to NumPy and Other Frameworks
 ========================================
 
 MLX array implements the `Python Buffer Protocol <https://docs.python.org/3/c-api/buffer.html>`_.
+
```
Comment (awni) [inline docs/src/usage/numpy.rst:7]:
```suggestion

```

File: docs/src/usage/numpy.rst [hunk 4]
```diff
@@ -3,7 +3,11 @@
 Conversion to NumPy and Other Frameworks
 ========================================
 
-MLX array implements the `Python Buffer Protocol <https://docs.python.org/3/c-api/buffer.html>`_.
+MLX array supports conversion between other frameworks through either:  
+
+1. The `Python Buffer Protocol <https://docs.python.org/3/c-api/buffer.html>`_. 
+2. `DLPack <https://dmlc.github.io/dlpack/latest/>`_.
```
Comment (awni) [inline docs/src/usage/numpy.rst:9]:
```suggestion
MLX array supports conversion between other frameworks with either:  

* The `Python Buffer Protocol <https://docs.python.org/3/c-api/buffer.html>`_. 
* `DLPack <https://dmlc.github.io/dlpack/latest/>`_.  
```

**Review by awni — APPROVED** [review]

**Pull request merged**